### DISCUSSION TITLE
include csrf token when doing http testing request

### DISCUSSION THREAD

**Discussion by mortenko** [adonisjs/core#1322] [Help] [answered]
Hi, is there any possibility to attach csrf token when making http post test request  ?  I'm getting actually invalid csrf token response from server. 

I just found solution to test it against current env like 
```js
csrf: {
enable: Env.get('NODE_ENV') === 'testing' ? false : true
.....
}
```
Its right way ? 

Here is my code:

```js
trait('Test/ApiClient');

test('some test', async({client}) => {
  const response = await client
      .post('/request')
      .attach('excel_file', Helpers.tmpPath('../test/excel_file.xlsx'))
      .end();
)
```
Thanks

**Comment by crbast** [comment, marked as answer]
hi @mortenko 👋
#### I think you can just disable CSRF for testing

> Its right way ?

Yep

#### Get CSRF token in controllers : 
```js
request.csrfToken
```
*source :  https://forum.adonisjs.com/t/how-to-access-csrf-token-from-controller/698*
